File: src/cn/timelives/java/math/algebra/abstractAlgebra/calculator/FieldCalculator.java
/**
 * 2018-02-28
 */
package cn.timelives.java.math.algebra.abstractAlgebra.calculator;

/**
 * A field calculator is 
 * @author liyicheng
 * 2018-02-28 19:29
 *
 */
public interface FieldCalculator<T> extends DivisionRingCalculator<T> {
	/**
	 * Returns the result of {@code x*y}, which should be commutative.
	 */
	@Override
	T multiply(T x, T y);
	
	
}

File: src/cn/ancono/math/algebra/abstractAlgebra/calculator/FieldCalculator.kt
/**
 * 2018-02-28
 */
package cn.ancono.math.algebra.abstractAlgebra.calculator

/**
 * A calculator for a field.
 * @author liyicheng
 * 2018-02-28 19:29
 */
interface FieldCalculator<T : Any> : DivisionRingCalculator<T> {
    /**
     * Returns the result of `x*y`, which should be commutative.
     */
    override fun multiply(x: T, y: T): T

    override val multiplyIsCommutative: Boolean
        get() = true

}
